R/calcFeDemandIndustry.R

```
@@ -21,10 +23,26 @@
 #' @importFrom tidyselect all_of
 #' @author Michaja Pehl
 #'
-calcFeDemandIndustry <- function(use_ODYM_RECC = FALSE, last_empirical_year = 2020) {
+calcFeDemandIndustry <- function(scenario, use_ODYM_RECC = FALSE, last_empirical_year = 2020) {
+
+  # Replace calls to SSPs and SSP2IndiaDEAs to individual scenarios, if present
+  if ("SSPs" %in% scenario) {
+    scenario <- c(scenario[!grepl("SSPs", scenario)], c("SSP1", "SSP2", "SSP3", "SSP4", "SSP5"))
+  }
+  if ("SSP2IndiaDEAs" %in% scenario) {
+    scenario <- c(scenario[!grepl("SSP2IndiaDEAs", scenario)], c("SSP2IndiaMedium", "SSP2IndiaHigh"))
+  }
+
+  remind_scenarios <- c(
+    paste0("SSP", c(1:5, "2_lowEn", "2_highDemDEU", "2IndiaHigh", "2IndiaMedium")),
+    paste0("SDP", c("", "_EI", "_RC", "_MC"))
+  )
```

Sorry for not suggesting this earlier, but could you throw an error here if any of `scenario` is not in `remind_scenarios`?

---

Thanks!

---

Instead of throwing an error, we could duplicate a default scenario, say SSP2, and add it under the unknown scenario's name. Something like this at the end? 
``` 
# Create data for any missing scenarios (i.e. not in remind) by duplication of the SSP2 data.
  remindAdd <- purrr::map(scenario[! scenario %in% getNames(remind, dim = "scenario")], function(scen) {
    message(glue::glue("Adding {scen} FeDemandIndustry scenario as copy of SSP2."))
    setItems(remind[, , "SSP2"], 3.1, scen)
  }) %>%
    mbind()
  # Filter by scenario
  remind <- mbind(remind, remindAdd) %>% mselect(scenario = subset)
```
(We do it anyways for SSP2_NAV_all...)